frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -56,63 +59,69 @@ function ReaderTypeset:onReadSettings(config)
     -- Block rendering mode: stay with legacy rendering for books
     -- previously opened so bookmarks and highlights stay valid.
     -- For new books, use 'web' mode below in BLOCK_RENDERING_FLAGS
-    local block_rendering_default_mode = 3
-    self.block_rendering_mode = config:readSetting("copt_block_rendering_mode")
-    if not self.block_rendering_mode then
-        if config:readSetting("last_xpointer") then
+    if config:has("copt_block_rendering_mode") then
+        self.block_rendering_mode = config:readSetting("copt_block_rendering_mode")
+    else
+        if config:has("last_xpointer") then
             -- We have a last_xpointer: this book was previously opened
             self.block_rendering_mode = 0
         else
+            -- For new books, use 'web' mode
             self.block_rendering_mode = G_reader_settings:readSetting("copt_block_rendering_mode")
-                                            or block_rendering_default_mode
+                                     or 3
```

Comment:
`-- For new books, use the global setting, or 'web' mode if not set`
or move that comment after the `or 3 -- default to 'web' mode`

Reply:
Yeah, I wasn't sure quite what to do with those comments, thanks ;).